netplugin/netd.go

```diff
@@ -464,28 +465,36 @@ func handleStateEvents(netPlugin *plugin.NetPlugin, crt *crt.CRT, opts cliOpts,
 	go handleEndpointEvents(netPlugin, crt, opts, retErr)
 }
 
-func handleEvents(netPlugin *plugin.NetPlugin, crt *crt.CRT, opts cliOpts) error {
-
-	recvErr := make(chan error, 1)
+func handleEvents(netPlugin *plugin.NetPlugin, crt *crt.CRT, opts cliOpts) {
+	recvErr := make(chan error)
 
 	//monitor and process state change events
 	go handleStateEvents(netPlugin, crt, opts, recvErr)
+	startDockerEventPoll(netPlugin, crt, recvErr, opts)
+
+	go func() {
+		for {
+			err := <-recvErr
```

Comment:
Yeah I wasn’t quite sure how you wanted this to be handled. I’ll try and refresh the PR this evening, if not, tomorrow.

-Erik

> On Jun 8, 2015, at 5:30 PM, Madhav Puri <notifications@github.com <mailto:notifications@github.com>> wrote:
> 
> In netplugin/netd.go <https://github.com/contiv/netplugin/pull/92#discussion_r31973889>:
> 
> >  
> >  	//monitor and process state change events
> >  	go handleStateEvents(netPlugin, crt, opts, recvErr)
> > +	startDockerEventPoll(netPlugin, crt, recvErr, opts)
> > +
> > +	go func() {
> > +		for {
> > +			err := <-recvErr
> I think this will cause startDockerEventPoll() to be called even if the error was set by handleStateEvents() in which case we should probably exit. May be we can create a separate error channel for startDockerEventPoll() to signal errors?
> 
> Also even when startDockerEventPoll() errors out may be we shall check for a specific error (like broken pipe or something more specific that can indicate a docker daemon restart) and continue to exit in other scenarios.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/contiv/netplugin/pull/92/files#r31973889>.
>